frontend/view-edit-lead/lead-home/view-edit-lead-contact-info.component.tsx

```diff
@@ -1,35 +1,126 @@
 import React from "react";
+import easyFetch from "../../util/easy-fetch";
 import { SingleLead } from "../view-lead.component";
 import { formatPhone } from "../../util/formatters";
 import LeadSection from "./lead-section.component";
+import LeadContactInformationInputs from "../edit-lead-inputs/lead-contact-information-inputs.component";
 
 export default function ViewEditLeadContactInfo(
   props: ViewEditLeadContactInfoProps
 ) {
+  const [editing, setEditing] = React.useState(false);
+  const [apiStatus, dispatchApiStatus] = React.useReducer(updatingReducer, {
+    isUpdating: false,
+    newLeadData: null
+  });
   const { lead } = props;
 
+  React.useEffect(() => {
+    if (apiStatus.isUpdating) {
+      const abortController = new AbortController();
+      easyFetch(`/api/leads/${lead.id}`, {
+        method: "PATCH",
+        body: apiStatus.newLeadData,
+        signal: abortController.signal
+      })
+        .then(data => {
+          dispatchApiStatus({ type: "reset" });
+          props.leadUpdated(data.lead);
+          setEditing(false);
```

Comment:
Why not put `editing` into the reducer state, so that `dispatchApiStatus({ type: "reset" })` can update the editing property itself. That would avoid potential double re-render due to calling setState twice.

Reply:
If the fix I put it is correct I'll add the same change to the basic lead info.